examples/dwarfdump.rs

```diff
@@ -647,7 +682,40 @@ where
         )?;
     }
     if flags.info {
-        dump_info(&dwarf, flags)?;
+        let dwo_parent_headers = if let Some(loader) = load_dwo_parent_section.as_mut() {
+            let dwarf = gimli::Dwarf::load(loader)?;
+            match dwarf
+                .units()
+                .map(|unit_header| dwarf.unit(unit_header))
+                .filter_map(|unit| {
+                    let dwo_id = match unit.header.type_() {
+                        gimli::UnitType::Skeleton(dwo_id) => dwo_id,
+                        gimli::UnitType::Compilation => {
+                            let mut entries = unit.entries();
+                            let (_, this) = entries.next_dfs()?.unwrap();
+                            match this.attr_value(gimli::constants::DW_AT_GNU_dwo_id)? {
+                                Some(gimli::AttributeValue::DwoId(dwo_id)) => dwo_id,
+                                Some(_) => unreachable!(),
```

Comment:
What makes this unreachable? (and same for the other use below)

Reply:
`dwoid!` only ever returns `AttributeValue::DwoId`

Reply:
Right, but then it falls through to `self.value.clone()` for other forms.